test/integration/protocol_integration_test.cc

```diff
@@ -2687,4 +2687,40 @@ TEST_P(ProtocolIntegrationTest, ReqRespSizeStats) {
   test_server_->waitUntilHistogramHasSamples("cluster.cluster_0.upstream_rs_headers_size");
 }
 
+TEST_P(ProtocolIntegrationTest, ResetLargeResponseUponReceivingHeaders) {
+  if (downstreamProtocol() == Http::CodecType::HTTP1) {
+    return;
+  }
```

Comment:
We should not exclude H1 downstreams.

Reply:
The flow control in http2_options doesn't apply to H1. So this test can't make the codec to send the response header but buffer some of the response body. Unless there are other H1 flow control options?